code/modules/mob/living/carbon/hallucination.dm

```diff
@@ -0,0 +1,1102 @@
+
+///////////////////////////////////////Hallucinated Hearing///////////////////////////
+/mob/living/carbon/hear_say(var/message, var/verb = "says", var/datum/language/language = null, var/alt_name = "",var/italics = 0, var/mob/speaker = null, var/sound/speech_sound, var/sound_vol)
+	if(hallucination >= 60 && prob(1))
+		var/orig_message = message
+		message = LAZYPICK(hallucinated_phrases, message)
+		log_say("Hallucination level changed [orig_message] by [speaker] to [message] for [key_name(src)].", ckey=key_name(src))
+	..()
+/mob/living/carbon/hear_radio(var/message, var/verb="says", var/datum/language/language=null, var/part_a, var/part_b, var/mob/speaker = null, var/hard_to_hear = 0, var/vname ="")
+	if(hallucination >= 60 && prob(1))
+		var/orig_message = message
+		message = LAZYPICK(hallucinated_phrases, message)
+		log_say("Hallucination level changed [orig_message] by [speaker] to [message] for [key_name(src)].", ckey=key_name(src))
+	..()
+
+/mob/living/carbon/proc/handle_hallucinations()     //Main handling proc, called in life()
+
+	hallucination -= 1	//Tick down the duration
+
+	if(!hallucination)  //We're done
+		return
+
+	if(!client || stat || world.time < next_hallucination)
+		return
+
+	var/hall_delay = rand(160,250)		//Time between hallucinations, modified by switch below.
+
+	switch(hallucination)	//26-74 are intentionally left off, as they do not modify the delay. This is a pretty common range for hallucinations.
+		if(1 to 25)		//Winding down, less frequent.
+			hall_delay *= 2
+		if(75 to 399)		//Yo mind really fucked, more frequent.
+			hall_delay *= 0.75
+		if(400 to INFINITY)		//This should only be possible in cult conversions. Very low delay to represent your flayed mind.
+			hall_delay *= 0.25
+
+	next_hallucination = world.time + hall_delay
+	var/list/candidates = list()
+	for(var/T in subtypesof(/datum/hallucination/))
+		var/datum/hallucination/H = new T
+		if(H.can_affect(src))
+			candidates += H
+	if(candidates.len)
+		var/datum/hallucination/H = pick(candidates)
+		H.holder = src
+		H.activate()
+
+
+///////////////////////////Hallucination Datums//////////////////////////
+
+/datum/hallucination
+	var/mob/living/carbon/holder	//Who is hallucinating?
+	var/allow_duplicates = TRUE     //This is set to false for hallucinations with long durations or ones we do not want repeated for a time
+	var/duration = 10				//how long before we call end()
+	var/min_power = 0 				//mobs only get this hallucination at this threshold
+	var/max_power = INFINITY		//mobs don't get this hallucination if it's above this threshold. Used to weed out more common ones if you're super fucked up
+	var/hearing_dependent = FALSE	//Deaf people won't get these if true
+
+///////////Things you involuntarily emote while hallucinating //////////////
+	var/list/hal_emote = list("mutters quietly.", "stares.", "grunts.", "looks around.", "twitches.", "shivers.", "swats at the air.", "wobbles.", "gasps!", "blinks rapidly.", "murmurs.",
```

Comment:
```suggestion
	var/static/list/hal_emote = list("mutters quietly.", "stares.", "grunts.", "looks around.", "twitches.", "shivers.", "swats at the air.", "wobbles.", "gasps!", "blinks rapidly.", "murmurs.",
```

Reply:
Also why is this stuff on the datum? This looks like a global list kinda thing. Maybe this should be in the same subsystem as the hallucination datum list.